docs/rfcs/0004-cirrus-experiments.md

```diff
@@ -0,0 +1,260 @@
+
+    1.  [Jared Lockhart](https://github.com/jaredlockhart)  The monitor A/A has now launched! 🎉 We expect to wrap it at the end of October, and barring any significant unforeseen complications, we would expect to then move on to focusing on the VPN integration.  Until then the VPN team can get a head start by beginning Glean integration.  We'll only need a single test event from each app to do a similar A/A validation for VPN so it's not necessary to rigorously instrument all of the VPN applications with Glean to test.
+
+    2.  [Beatriz Rizental](https://github.com/brizental) We are lucky that the VPN is already instrumented with Glean :) And we have more than one event.
+
+2.  How would Nimbus handle the case when users are assigned a random UUID before logging in? Is that the correct approach for such situations? What if the user is in an experiment branch that is conflicting with whatever experiment branch their user id is assigned to? Is this a concern or is it fine to just switch the user to the other branch after login?
+
+    1.  [Jared Lockhart](https://github.com/jaredlockhart) This is still an open question.  A naive strategy is that yes, when a user passes through an auth boundary and their client id changes, their feature configurations will be recomputed with their authenticated id.  I expect we will start there and then evaluate whether a more robust solution is necessary.
+
+3.  If a user opts out of telemetry/Glean, does this automatically opt them out of Experiments as well? How should we handle cases where the user is enrolled in an experiment at the time of opt out?  
+
+    1.  [Beatriz Rizental](https://github.com/brizental)  My hunch here is that users that have telemetry turned off will not participate in experiments. Wouldn't that break the targeting logic for example when it's based on percentage of users? However if a user is already enrolled in some experiment, we will keep them in that branch until the experiment is over. Otherwise turning data collection off would have unexpected behavior in some cases.
+
+    2.  [Jared Lockhart](https://github.com/jaredlockhart)  Our existing process in other applications is that at the time the user opts out of telemetry and/or experimentation (opting out of telemetry also opts out of experiments, opting out of experiments does not opt out of telemetry), they are unenrolled from the experiment. This can not be managed by the Nimbus infrastructure, it must be implemented by the integration application.  Ie there must be some UI that allows a user to opt out of telemetry and/or experimentation, the application must store that preference, and any application that is opted out must no longer send requests to Cirrus.
```

Comment:
I think this should be sketched out in this doc, as it add some complexity.

Reply:
I agree. I will just talk to Santiago about what he feels is the best approach, because this is a product question IMO.